front/components/data_source/DocumentOrTableUploadOrEditModal.tsx

```diff
@@ -0,0 +1,528 @@
+  contentNode,
+  isOpen,
+  onClose,
+  owner,
+  plan,
+}: DocumentOrTableUploadOrEditModalProps) {
+  const sendNotification = useContext(SendNotificationsContext);
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  const [tableOrDoc, setTableOrDoc] = useState<TableOrDocument>({
+    name: "",
+    description: "",
+    file: null,
+    text: "",
+    tags: [],
+    sourceUrl: "",
+  });
+
+  const [uploading, setUploading] = useState(false);
+  const [isBigFile, setIsBigFile] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [developerOptionsVisible, setDeveloperOptionsVisible] = useState(false);
+
+  const isTable = contentNode?.type === "database";
+  const initialId = contentNode?.internalId;
+
+  const { table } = useTable({
+    workspaceId: owner.sId,
+    dataSourceName: dataSourceView.dataSource.name,
+    tableId: isTable ? initialId ?? null : null,
+  });
+
+  const resetTableOrDoc = () => {
+    setTableOrDoc({
+      name: "",
+      description: "",
+      file: null,
+      text: "",
+      tags: [],
+      sourceUrl: "",
+    });
+  };
+  // TODO(GROUPS_UI) replace endpoint https://github.com/dust-tt/dust/issues/6921
+  useEffect(() => {
+    if (initialId) {
+      setIsLoading(true);
+      if (table) {
+        setTableOrDoc((prev) => ({
+          ...prev,
+          name: table.name,
+          description: table.description,
+        }));
+      } else {
+        setUploading(true);
+        void (async () => {
+          try {
+            const res = await fetch(
+              `/api/w/${owner.sId}/data_sources/${
+                dataSourceView.dataSource.name
+              }/documents/${encodeURIComponent(initialId)}`
+            );
+            if (res.ok) {
+              const document = await res.json();
+              setTableOrDoc((prev) => ({
+                ...prev,
+                name: initialId,
+                text: document.document.text,
+                tags: document.document.tags,
+                sourceUrl: document.document.source_url,
+              }));
+            }
+          } catch (e) {
+            console.error(e);
+          } finally {
+            setUploading(false);
+          }
+        })();
+      }
+      setIsLoading(false);
+    } else {
+      resetTableOrDoc();
+    }
+  }, [isTable, initialId, table, owner.sId, dataSourceView.dataSource.name]);
+
+  const total = 0; // TODO: Get the total number of documents
+
+  if (
+    !initialId &&
+    plan.limits.dataSources.documents.count !== -1 &&
+    total >= plan.limits.dataSources.documents.count
+  ) {
+    return (
+      <DocumentLimitPopup
+        isOpen={isOpen}
+        plan={plan}
+        onClose={() => onClose(false)}
+        owner={owner}
+      />
+    );
+  }
+
+  const handleTableUpload = async () => {
+    setUploading(true);
+    try {
+      const fileContent = tableOrDoc.file
+        ? await handleFileUploadToText(tableOrDoc.file)
+        : null;
```

Comment:
We should not need this for table upload.

Reply:
We actually convert the file to text and then convert it to a csv format. 

See current implementation [here](https://github.com/dust-tt/dust/blob/0bc1d6499590952046b8c093b6d17be824b12292/front/components/data_source/TableUploadOrEditModal.tsx#L104)